README.md

# eeiR - Ecological Evaluation Index


<p align="center">
<img src="https://github.com/salah31416/eeiR/raw/master/inst/figures/macro.png" 
alt="Classes" width="35%"/>
</p>

[![DOI](https://zenodo.org/badge/185321992.svg)](https://zenodo.org/badge/latestdoi/185321992)


## Contents

* [Installation](#installation)
* [Ecological Status Group - ESG](#ecological-status-group)
* [Ecological Evaluation Index - EEI-c](#ecological-evaluation-index)
* [Ecological Status Class - ESC](#ecological-status-class)
* [Ecological Quality Ratio - EQR](#ecological-quality-ratio)
* [Graphics](#graphics)


# Installation

```r
install.packages("devtools")

devtools::install_github("salah31416/eeiR")
```

## Ecological Status Group

<img src="https://latex.codecogs.com/svg.latex?\Large&space;ESG%20I%20(\%%20coverage)%20=%20(IA%20\cdot%201)%20+%20(IB%20\cdot%200.8)%20+%20(IC%20\cdot%200.6)" title="ESGI" />

<img src="https://latex.codecogs.com/svg.latex?ESG%20II%20(\%%20coverage)%20=%20(IIA%20\cdot%200.8)%20+%20(IIB%20\cdot%201)%20+%20(IIC%20\cdot%201)" title=ESG2 />


```r
## %Coverage ESG I
ia = c(0, 10, 5, 0, 0)
ib = c(25, 27, 13, 15, 12)
ic = c(18, 12, 12, 6.1, 5.1)

## %Coverage ESG II
iia = c(2.2, 5, 6.55, 55.05, 60.05)
iib = c(69.85, 43.1, 36.85, 34.4, 55)

## ESG I
esg1 = esg("1", ia, ib, ic)
#[1] 30.80 38.80 22.60 15.66 12.66

## ESG II
esg2 = esg("2", iia, iib, 0)
#[1]  71.61  47.10  42.09  78.44 103.04
```

## Ecological Evaluation Index

<img src="https://latex.codecogs.com/svg.latex?x%20=%20\frac{ESGI}{100};%20\quad%20y%20=%20\frac{ESGII}{100}" title="xy" />


<a href="https://www.codecogs.com/eqnedit.php?latex=EEI=\begin{cases}&space;hyp=a&plus;bx&plus;cx^2&plus;dy&plus;ey^2&plus;fxy&space;\\\\&space;ESI=\begin{cases}&space;10&space;&&space;\text{&space;if&space;}&space;hyp>1&space;\\&space;2&plus;8\cdot&space;hyp&&space;\end{cases}&space;\\\\&space;EEIc=\begin{cases}&space;2&space;&&space;\text{&space;if&space;}&space;ESI<2&space;\\&space;ESI&&space;\end{cases}&space;\end{cases}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?EEI=\begin{cases}&space;hyp=a&plus;bx&plus;cx^2&plus;dy&plus;ey^2&plus;fxy&space;\\\\&space;ESI=\begin{cases}&space;10&space;&&space;\text{&space;if&space;}&space;hyp>1&space;\\&space;2&plus;8\cdot&space;hyp&&space;\end{cases}&space;\\\\&space;EEIc=\begin{cases}&space;2&space;&&space;\text{&space;if&space;}&space;ESI<2&space;\\&space;ESI&&space;\end{cases}&space;\end{cases}" title="EEI=\begin{cases} hyp=a+bx+cx^2+dy+ey^2+fxy \\\\ ESI=\begin{cases} 10 & \text{ if } hyp>1 \\ 2+8\cdot hyp& \end{cases} \\\\ EEIc=\begin{cases} 2 & \text{ if } ESI<2 \\ ESI& \end{cases} \end{cases}" /></a>

where: *x* is the score in *ESG I*, *y* is the score in *ESG II*
and *a*, *b*, *c*, *d*, *e* and *f* are the coefficients of the hyperbola.

```r
## hyp
hyp = eei(esg1, esg2, "hyp")
#[1]  0.22081475  0.43121705  0.34064851  0.06959875 -0.04513925

## ESI
esi = eei(esg1, esg2, "esi")
#[1] 3.766518 5.449736 4.725188 2.556790 1.638886

## EEIc
eeic = eei(esg1, esg2, "eeic")
#[1] 3.766518 5.449736 4.725188 2.556790 2.000000
```

## Ecological Quality Ratio

<a href="https://www.codecogs.com/eqnedit.php?latex=EQR=1.25\cdot(EEIc/10)-0.25" target="_blank"><img src="https://latex.codecogs.com/gif.latex?EQR=1.25\cdot(EEIc/RC)-0.25" title="EQR=1.25\cdot(eeic/RC)-0.25" /></a>

Where: *RC* = 10

```r
## EQR
eqr(esi)
#[1]  0.22081475  0.43121705  0.34064851  0.06959875 -0.04513925

eqr(eeic)
#[1] 0.22081475 0.43121705 0.34064851 0.06959875 0.00000000
```

## Ecological Status Class

<a href="https://www.codecogs.com/eqnedit.php?latex=ESC=\begin{cases}&space;Bad&space;&&space;\text{&space;if&space;}&space;eeic\leq2&space;\\&space;Low&space;&&space;\text{&space;if&space;}&space;2<eeic\leq4&space;\\&space;Moderate&space;&&space;\text{&space;if&space;}&space;4<eeic\leq6&space;\\&space;Good&space;&&space;\text{&space;if&space;}&space;6<eeic\leq8&space;\\&space;High&space;&&space;\text{&space;if&space;}&space;eeic>8&space;\end{cases}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?ESC=\begin{cases}&space;Bad&space;&&space;\text{&space;if&space;}&space;eeic\leq2&space;\\&space;Low&space;&&space;\text{&space;if&space;}&space;2<eeic\leq4&space;\\&space;Moderate&space;&&space;\text{&space;if&space;}&space;4<eeic\leq6&space;\\&space;Good&space;&&space;\text{&space;if&space;}&space;6<eeic\leq8&space;\\&space;High&space;&&space;\text{&space;if&space;}&space;eeic>8&space;\end{cases}" title="ESC=\begin{cases} Bad & \text{ if } eeic\leq2 \\ Low & \text{ if } 2<eeic\leq4 \\ Moderate & \text{ if } 4<eeic\leq6 \\ Good & \text{ if } 6<eeic\leq8 \\ High & \text{ if } eeic>8 \end{cases}" /></a>

```r
## ESC
esc(eeic)
#[1] "Low"     "Moderate" "Moderate" "Low"     "Bad"
```

## Graphics

<p align="center">
<img src="https://github.com/salah31416/eeiR/raw/master/inst/figures/classes.png" alt="Classes" width="50%"/>
<img src="https://github.com/salah31416/eeiR/raw/master/inst/figures/hyp.png" alt="Hyperbole" width="80%"/>
<img src="https://github.com/salah31416/eeiR/raw/master/inst/figures/k2.png" alt="Hyperbole 3D" width="60%"/>
</p>

# References

> Orfanidis, S., Panayotidis, P., & Stamatis, N. (2003). An insight to the ecological evaluation index (EEI). Ecological Indicators, 3(1), 27–33. doi:10.1016/s1470-160x(03)00008-6

> Orfanidis, S., Pinna, M., Sabetta, L., Stamatis, N., & Nakou, K. (2008). Variation of structural and functional metrics in macrophyte communities within two habitats of eastern Mediterranean coastal lagoons: natural versus human effects. Aquatic Conservation: Marine and Freshwater Ecosystems, 18(S1), S45–S61. doi:10.1002/aqc.957

> Orfanidis, S., Panayotidis, P., & Ugland, K. (2011). Ecological Evaluation Index Continuous Formula (EEI-c) Application: A Step Forward For Functional Groups, The Formula And Reference Condition Values. Mediterranean Marine Science, 12(1), 199. doi:10.12681/mms.60

> Orfanidis, S., Dencheva, K., Nakou, K., Tsioli, S., Papathanasiou, V., & Rosati, I. (2014). Benthic macrophyte metrics as bioindicators of water quality: towards overcoming typological boundaries and methodological tradition in Mediterranean and Black Seas. Hydrobiologia, 740(1), 61–78. doi:10.1007/s10750-014-1938-x

> Caldeira, A. Q., De Paula, J. C., Reis, R. P., & Giordano, R. G. (2017). Structural and functional losses in macroalgal assemblages in a southeastern Brazilian bay over more than a decade. Ecological Indicators, 75, 242–248. https://doi.org/10.1016/j.ecolind.2016.12.029

> Caldeira, A. Q., & Reis, R. P. (2019). Brazilian macroalgae assemblages analyzed using the ecological evaluation index (EEI-c). Ocean & Coastal Management, 104927. https://doi.org/10.1016/j.ocecoaman.2019.104927
 


NEWS.md
# eeiR 0.0.1.020

	2020-01-06 Package release.



R/eei.R

##-------------------------------------------------------------
## Ecological Quality Ratios - EQR
##-------------------------------------------------------------
eqr = function(x) 
{
	rc = 10
	out = 1.25 * (x/rc) - 0.25

	return(out)
}#end eqr

##-------------------------------------------------------------
## Ecological Status Classes - ESC
##-------------------------------------------------------------
esc = function(x, FUN = NULL)
{
	if(is.null(FUN)) {
		FUN = function(ic) {
			if(ic <= 2) cl = "Bad"
			if(ic > 2 & ic <= 4) cl = "Poor"
			if(ic > 4 & ic <= 6) cl = "Moderate"
			if(ic > 6 & ic <= 8) cl = "Good"
			if(ic > 8) cl = "High"
			return(cl)
		}#end FUN
	}#end if

	return(sapply(x, FUN))
}#end esc

##-------------------------------------------------------------
## Ecological Status Group - ESG
##-------------------------------------------------------------
esg = function(ia = 0, ib = 0, ic = 0, k = NULL, type = c("esg1", "esg2")) 
{
	type = match.arg(type)

	if(is.null(k) & type == "esg1") k = c(1, 0.8, 0.6)
	if(is.null(k) & type == "esg2") k = c(0.8, 1, 0)

	out = ia * k[1] + ib * k[2] + ic * k[3]

	return(out)
}#end esg

##-------------------------------------------------------------
## Ecological Evaluation Index - EEI
##-------------------------------------------------------------
eei_hyp = function(x, y, k = c(a = 0.503, b = 0.954, c = -0.204, d = -0.998, e = 0.355, f = -0.109)) 
{
	x = x/100
	y = y/100

	#     a  +   bx   +   cx^2   +   dy   +   ey^2   +   fxy
	#     |	     |        |          |        |          |
	h = k[1] + k[2]*x + k[3]*x^2 + k[4]*y + k[5]*y^2 + k[6]*x*y

	h[h > 1] <- 1

	## Ecological Evaluation Index
	eei = 2 + 8 * h

	return(eei)
}#end hyp

##-------------------------------------------------------------
## Ecological Evaluation Index Coverage - EEIc
##-------------------------------------------------------------
eeic = function(x, y, k = c(a = 0.503, b = 0.954, c = -0.204, d = -0.998, e = 0.355, f = -0.110)) 
{
    ## EEI
    h = eei_hyp(x, y, k)

	## eeic
	h[h < 2] <- 2
    
    return(h)
}#end eeic

##-------------------------------------------------------------
## Generate Sequence Number - gsn
##-------------------------------------------------------------
.gsn = function(RV = TRUE, from = 0, to = 100, by = .5) 
{
	ESC=V=R=z=D3=EEIc=NULL

	x = y = seq(from, to, by)
	sq = expand.grid(x = x, y = y)
	SEQ = data.table(sq)


	if(isTRUE(RV) | is.data.table(RV))
	{
		if(isTRUE(RV))
		{
			RV = data.table(
			  R = c(
				"x <= 30 & y > 60",
				"x <= 30 & (y > 30 & y <= 60)",
				"(x > 30 & x <= 60) & y > 60",
				"x > 60 & y > 60",
				"(x > 30 & x <= 60) & (y > 30 & y <= 60)",
				"x <= 30 & y <= 30",
				"x > 60 & (y > 30 & y <= 60)",
				"(x > 30 & x <= 60) & y <= 30",
				"x > 60 & y <= 30"),
			  V = c(2, 4, 4, 6, 6, 6, 8, 8, 10 ),
			  ESC = c("Bad", "Poor", "Poor", "Moderate", "Moderate",
					  "Moderate", "Good", "Good", "High") )
		}#end if

		message("Classification of intervals")
		cat(paste(ESC, V, R), sep="\n")

		for(i in 1:nrow(RV))
		{
			rela = parse(text=RV[i,1])
			SEQ[eval(rela), z := RV[i,2]]

#			z0 = SEQ[eval(rela), unique(z)]
#			w = paste(paste0(" ", z0," ", RV[i,3], ":"),
#					  RV[i,1], sep="  ")
#			message(w)
		}#end for
	}#end if

	return(SEQ[])
}#end generate_seq

